File: Unity3D/VisualesUnity/Assets/Scripts/OSCListener.js
private var UDPHost : String = "127.0.0.1";
private var listenerPort : int = 10001;
private var broadcastPort : int = 10001;
private var oscHandler : Osc;

private var eventName : String = "";
private var eventData : String = "";
private var counter : int = 0;
public var output_txt : GUIText;


public function Start ()
{	
	var udp : UDPPacketIO = GetComponent("UDPPacketIO");
	udp.init(UDPHost, broadcastPort, listenerPort);
	oscHandler = GetComponent("Osc");
	oscHandler.init(udp);
			
	oscHandler.SetAddressHandler("/updateBlob", updateText);
	
}
Debug.Log("Running");

function Update () {
	output_txt.text = "Event: " + eventName + " Event data: " + eventData;
	
	var cube = GameObject.Find("Cube");
	var boxWidth:int = counter;
    cube.transform.localScale = Vector3(boxWidth,5,5);	
}	

public function updateText(oscMessage : OscMessage) : void
{	
	eventName = Osc.OscMessageToString(oscMessage);
	eventData = oscMessage.Values[0];
} 

File: Unity3D/VisualesUnity/Assets/Scripts/OSCListener.cs
using UnityEngine;
using System.Collections;

public class OSCListener : MonoBehaviour {
	public static OSCListener instance;
	public string UDPHost = "127.0.0.1";
	public int listenerPort = 10001;
	public int broadcastPort = 10001;

	Osc oscHandler;
	public static OscMessage updateBlobMessage = null;
	public static OscMessage removeBlobMessage = null;
	bool newUpdateBlobMessageFrame = false;
	bool newRemoveBlobMessageFrame = false;
	OscMessage newUpdateBlobMessage = null;
	OscMessage newRemoveBlobMessage = null;
	void Awake(){
		instance = this;

	}
	void Start () {
		UDPPacketIO udp = GetComponent<UDPPacketIO>();
		udp.init(UDPHost,broadcastPort,listenerPort);
		oscHandler = GetComponent<Osc>();
		oscHandler.init(udp);
		oscHandler.SetAddressHandler("/updateBlob",UpdateBlob);
		oscHandler.SetAddressHandler("/removeBlob",RemoveBlob);

	}

	void Update(){
		if(newUpdateBlobMessageFrame){
			OSCListener.updateBlobMessage = newUpdateBlobMessage;
			newUpdateBlobMessageFrame = false;
		} else {
			OSCListener.updateBlobMessage = null;
		}

		if(newRemoveBlobMessageFrame){
			OSCListener.removeBlobMessage = newRemoveBlobMessage;
			newRemoveBlobMessageFrame = false;
		} else {
			OSCListener.removeBlobMessage = null;
		}

	}
	void UpdateBlob(OscMessage message){
		newUpdateBlobMessageFrame = true;
		newUpdateBlobMessage = message;
	}
	void RemoveBlob(OscMessage message){
		newRemoveBlobMessageFrame = true;
		OSCListener.removeBlobMessage = message;
	}
}
